core/formats/formats_10.cpp

```diff
@@ -3230,6 +3397,9 @@ void postings_reader_base::prepare(index_input& in, const reader_state& state,
                    "invalid block size '",
                    block_size, "', expected '", block_size_, "'")};
   }
+
+  scorers_ =
+    state.scorers.subspan(0, std::min(state.scorers.size(), kMaxScorers));
```

Comment:
Maybe better to make precondition with size <= kMaxScorers?
I think it's more clear than just cut tail